File: Search/JumpSearch/jumpSearch.py
import math
def jump_search(arr,search):

    j_block = len(arr)
    low = 0
    interval = int(math.sqrt(j_block))
    for i in range(0,j_block,interval):
        if arr[i] < search:
            low = i
        elif arr[i] == search:
            return i
        else:
            break # bigger number is found
    c=low
    for j in arr[low:]:
        if j==search:
            return c
        c+=1
    return "Not found"

arr = [ 0, 1, 1, 2, 3, 5, 8, 13, 21,34, 55, 89, 144, 233, 377, 610 ] 

search_block = 55    
res = jump_search(arr, search_block)
print("Number" , search_block, "is at index" ,"%.0f"%res)

File: Search/JumpSearch/jumpSearch.cpp
#include<bits/stdc++.h> //you can also use #include<iostream.h> but then also include math.h
using namespace std;
#define MAX 500
//function to perfrom jump search
int jumpSearch(int array[],int x,int n)
{
  //Finding block size to be jumped
  int st=sqrt(n);
  //Finding the block where the element is present
  int prev=0;
  while(array[min(st,n)-1]<x)
  {
    prev=st;
    st+=sqrt(n);
    if(prev>=n)
       return -1;
  }
  //Doing liner search to find the element
  while(array[prev]<x)
  {
    prev++;
    if(prev==min(st,n)) // if we reach the next block
      return -1;
  }
  //else if the element is found
  if(array[prev]==x)
  {
    return prev;
  }
  return -1;
}
int main()
{
  int arr[MAX];
  cout<<"Enter the number of array elements: ";
  int n,search;
  cin>>n;
  for(int i=0;i<n;i++)   //Taking input to array
    cin>>arr[i];
  cout<<"Enter the element to be searched: ";
  cin>>search;
  int index=jumpSearch(arr,search,n);
  //print the index where the element is located
  if(index==-1)
  cout<<"Element not found";
  else
  cout<<"The element is at index "<<index;

  return 0;

}
